panedr/tests/test_edr.py

```diff
@@ -182,35 +171,68 @@ def test_progress(self):
         print(df.iloc[-1, 0])
         ref_line = 'Last Frame read : 14099, time : 28198000.0 ps'
         last_line = next(progress_iter)
-        self.assertEqual(ref_line, last_line)
+        assert ref_line == last_line
         # Did we leave stderr clean with a nice new line at the end?
-        self.assertTrue(output.getvalue().endswith('\n'),
-                        'The new line is missing at the end.')
+        assert output.getvalue().endswith('\n'), \
+               'New line missing at the end of output.'
 
     def _assert_progress_range(self, progress, dt, start, stop, step):
         for frame_idx in range(start, stop, step):
             ref_line = 'Read frame : {},  time : {} ps'.format(frame_idx,
                                                                dt * frame_idx)
             progress_line = next(progress)
             print(frame_idx, progress_line)
-            self.assertEqual(ref_line, progress_line)
+            assert ref_line == progress_line
 
 
 def read_xvg(path):
     """
-    Read a XVG file and returns a 2D numpy array.
+    Reads XVG file, returning the data, names, and precision.
+    
+    The data is returned as a 2D numpy array. Column names are returned as an
+    array of string objects. Precision is an integer corresponding to the least
+    number of decimal places found, excluding the first (time) column.
 
     The XVG file type is assumed to be 'xy' or 'nxy'. The function also assumes
     that there is only one serie in the file (no data after // is // is
     present). If more than one serie are present, they will be concatenated if
     the number of column is consistent, is the number of column is not
     consistent among the series, then the function will crash.
     """
+    data = []
+    names = []
+    prec = -1
     with open(path) as infile:
-         data = numpy.array([line.split() for line in infile
-                             if not line[0] in '@#%&/'],
-                            dtype=float)
-    return data
+        for line in infile:
+            if not re.match(COMMENT_PATTERN, line):
+                data.append(line.split())
+                precs = [ndec(val) for val in data[-1][1:]]
+                if prec == -1:
+                    prec = min(precs)
+                else:
+                    prec = min(prec, *precs)
+                continue
+            mtx = re.match(LEGEND_PATTERN, line)
```

Comment:
`mtx` is not an explicit variable name, actually I do not see what it stands for. Could you find a more explicit name?